uber/models.py

```diff
@@ -759,35 +766,43 @@ def auto_badge_num(self, badge_type):
             a specific range.
             :return:
             """
-            in_range = self.query(Attendee.badge_num).filter(Attendee.badge_num >= c.BADGE_RANGES[badge_type][0],
-                                                             Attendee.badge_num <= c.BADGE_RANGES[badge_type][1])
-            if in_range.count():
-                in_range_list = [int(row[0]) for row in in_range.order_by(Attendee.badge_num).all()]
```

Comment:
Here we were performing TWO queries which would both result in a table scan, one right after the other. `in_range.count()` is as much of a performance hog as `in_range.all()`. Instead of calling `in_range.count()`, we are better off creating `in_range_list` first and calling `len()` on it, which is extremely fast.